src/modules/party/party.service.get-party-by-search-text.test.ts

```diff
@@ -0,0 +1,160 @@
+import { HttpService } from '@nestjs/axios';
+import { GetPartyGenerator } from '@ukef-test/support/generator/get-party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { AcbsGetPartiesBySearchTextResponse } from './dto/acbs-get-parties-by-search-text-response.dto';
+import { GetPartiesBySearchTextException } from './exception/get-parties-by-search-text.exception';
+import { PartyService } from './party.service';
+
+jest.mock('@ukef/modules/acbs/acbs-party.service');
+jest.mock('@ukef/modules/acbs/acbs-authentication.service');
+
+describe('PartyService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+
+  let httpService: HttpService;
+  let partyService: PartyService;
+
+  let httpServiceGet: jest.Mock;
+
+  beforeEach(() => {
+    httpService = new HttpService();
+
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    partyService = new PartyService({ baseUrl }, httpService, null, null);
+  });
+
+  describe('getPartyBySearchText', () => {
+    const getExpectedGetPartiesBySearchTextArguments = (searchText: string): [string, object] => [
+      `/Party/Search/${searchText}`,
+      {
+        baseURL: baseUrl,
+        headers: {
+          Authorization: `Bearer ${idToken}`,
+        },
+      },
+    ];
+
+    const { partiesInAcbs, parties } = new GetPartyGenerator(valueGenerator).generate({ numberToGenerate: 2 });
+    const partiesInAcbsWithPartyIdentifiers: AcbsGetPartiesBySearchTextResponse = [
+      { ...partiesInAcbs[0], PartyIdentifier: valueGenerator.stringOfNumericCharacters() },
+      { ...partiesInAcbs[1], PartyIdentifier: valueGenerator.stringOfNumericCharacters() },
+    ];
+
+    it('returns matching parties in the correct format if the request is successful', async () => {
```

Comment:
Can we please add some edge case assertions where `searchText` is following

* `   ` (3 white spaces)
* `A!@`